File: CSCI_313-323_Data_Structures_And_Algorithms/CSCI_313_Data_Structures/ExCode/11-OrderedSet-OrderedMap/LeetCode/LeetCode-2766.java
class Solution {
    /// O(nlogn) algorithm with O(n) extra space
    public List<Integer> relocateMarbles(int[] nums, int[] moveFrom, int[] moveTo) {
        TreeSet<Integer> marbles = new TreeSet<>();
        for (var num: nums) marbles.add(num);

        for (int i=0; i<moveFrom.length; i++){
            if (moveFrom[i] == moveTo[i]) continue;
            marbles.remove(moveFrom[i]);
            marbles.add(moveTo[i]);
        } // end-for

        return new ArrayList<Integer>(marbles);
    }
}

File: CSCI_313-323_Data_Structures_And_Algorithms/CSCI_313_Data_Structures/ExCode/11-OrderedSet-OrderedMap/LeetCode/LeetCode-2766.cpp
class Solution {
public:
    /// O(nlogn) algorithm with O(n) extra space
    vector<int> relocateMarbles(vector<int>& nums, vector<int>& moveFrom, vector<int>& moveTo) {
        set<int> marbles(nums.begin(), nums.end());
        for (int i=0; i<moveFrom.size(); i++){
            if (moveFrom[i] == moveTo[i]) continue;
            marbles.erase(moveFrom[i]);
            marbles.insert(moveTo[i]);
        } // end-for

        return vector<int>(marbles.begin(), marbles.end());
    } // end-relocateMarbles
};